shared/contracts/projects.contract.ts

```diff
@@ -49,6 +52,6 @@ export const projectsContract = contract.router({
     // TODO: we need to define filters, they should probably match filters for Projects. Or we might want to pass only project ids, which
     //       would be already filtered
     //query: z.object({ countryCodes: z.string().array().optional() }).optional(),
-    query: contract.type<{ filter: ProjectMapFilters }>(),
+    query: projectsQuerySchema.pick({ filter: true }),
```

Comment:
I specially like this, here we can just extract the 'sub-schema' from the main schema